mlrun/model.py

```diff
@@ -326,6 +326,44 @@ def iteration(self, iteration):
         self._iteration = iteration
 
 
+class HyperParamOptions(ModelObj):
+    """Hyper Parameter Options
+
+    Parameters:
+        param_file (str):       hyper params input file path/url, instead of inline
+        strategy (str):         hyper param strategy - grid, list or random
+        selector (str):         selection criteria for best result ([min|max.]<column>), e.g. max.accuracy
+        stop_condition (str):   early stop condition e.g. "accuracy > 0.9"
+        parallelism (int):      number of param combinations to run in parallel (over Dask)
+        dask_cluster_uri (str): db uri for a deployed dask cluster function, e.g. db://myproject/dask
+        max_evals (int):        max number of runs (in random strategy)
+        max_errors (int):       max number of child runs errors for the overall job to fail
+        teardown_dask (bool):   kill the dask cluster pods after the runs
+    """
+
+    def __init__(
+        self,
+        param_file=None,
+        strategy=None,
+        selector=None,
+        stop_condition=None,
+        parallelism=None,
+        dask_cluster_uri=None,
+        max_evals=None,
+        max_errors=None,
+        teardown_dask=None,
+    ):
+        self.param_file = param_file
+        self.strategy = strategy
+        self.selector = selector
+        self.stop_condition = stop_condition
+        self.max_evals = max_evals
```

Comment:
Create a setter for `max_evals` and explode when it's set and strategy is not `random`

Reply:
what if he sets the max eval first and the strategy later? isnt it better to validate in `generator`?